nginx.tmpl

```diff
@@ -0,0 +1,177 @@
+        }
+
+        {{with $mesos}}
+        location ~ /mesos/((master|metrics|monitor|files)\/.*) {
+            proxy_set_header host              $host;
+            proxy_set_header x-real-ip         $remote_addr;
+            proxy_set_header x-forwarded-for   $proxy_add_x_forwarded_for;
+            proxy_set_header x-forwarded-proto $scheme;
+
+            proxy_pass http://mesos/$1$is_args$args;
+        }
+
+        location /mesos {
+            root /usr/share/nginx/html;
+            index index.html index.htm;
+            try_files $uri $uri/ =404;
+        }
+
+        {{end}}
+
+        {{range service "follower.mesos"}}
+        location ~ /mesos/slave/{{index (.ID | split ":") 2}}/(.*) {
+            proxy_set_header        Host $host;
+            proxy_set_header        X-Real-IP $remote_addr;
+            proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header        X-Forwarded-Proto $scheme;
+            proxy_pass http://{{.Address}}:{{.Port}}/$1$is_args$args;
+        }{{end}}
+
+        {{with $marathon}}
+        location /marathon/ {
+            proxy_set_header        host $host;
+            proxy_set_header        x-real-ip $remote_addr;
+            proxy_set_header        x-forwarded-for $proxy_add_x_forwarded_for;
+            proxy_set_header        x-forwarded-proto $scheme;
+            proxy_pass http://marathon/;
+        } {{end}}
+
+        {{with $chronos}}
+        location /chronos/ {
+            proxy_set_header        host $host;
+            proxy_set_header        x-real-ip $remote_addr;
+            proxy_set_header        x-forwarded-for $proxy_add_x_forwarded_for;
+            proxy_set_header        x-forwarded-proto $scheme;
+            proxy_pass http://chronos/;
+        } {{end}}
+
+        {{with $consul}}
+        location ~ /consul/(v1\/.*) {
+            proxy_set_header        host $host;
+            proxy_set_header        x-real-ip $remote_addr;
+            proxy_set_header        x-forwarded-for $proxy_add_x_forwarded_for;
+            proxy_set_header        x-forwarded-proto $scheme;
+            proxy_pass {{if $consulSSL}}https{{else}}http{{end}}://consul/$1$is_args$args;
+        } {{end}}
+
+        location /consul {
+            root /usr/share/nginx/html;
+            index index.html index.htm;
+            try_files $uri $uri/ =404;
+        }
+
+        {{with $mantlapi}}
+        location /api/ {
+            proxy_set_header        host $host;
+            proxy_set_header        x-real-ip $remote_addr;
+            proxy_set_header        x-forwarded-for $proxy_add_x_forwarded_for;
+            proxy_set_header        x-forwarded-proto $scheme;
+            proxy_pass http://mantlapi/;
+        } {{end}}
+
+        {{with $traefikAdmin}}
+        location = /traefik {
+            return 301 /traefik/dashboard/;
+        }
+
+        location /traefik {
+            try_files $uri @traefik-upstream;
```

Comment:
@ryane this is the section (and important line) you're going to want to emulate for Mesos. Be aware that `proxy_pass` below in `location @traefik-upstream` must not have a path on it. That's what the rewrite is doing.